src/api/Intelligence.js

```diff
@@ -47,21 +43,15 @@ class Intelligence extends Base {
 
         const url = `${this.getBaseApiUrl()}/ai/ask`;
 
-        this.successCallback = successCallback;
-        this.errorCallback = errorCallback;
-
-        await this.xhr
-            .post({
-                url,
-                id: items[0].id,
-                data: {
-                    mode: 'single_item_qa',
-                    prompt,
-                    items,
-                },
-            })
-            .then(this.successHandler)
-            .catch(this.errorHandler);
+        return this.xhr.post({
```

Comment:
had to change this to return the the promise instead of resolving here and waiting for the callback